utils/src/main/kotlin/storage/XZCompressedLocalFileStorage.kt

```diff
@@ -33,7 +34,13 @@ class XZCompressedLocalFileStorage(
      */
     directory: File
 ) : LocalFileStorage(directory) {
-    override fun read(path: String) = XZCompressorInputStream(super.read("$path.xz"))
+    override fun read(path: String) =
+        try {
+            XZCompressorInputStream(super.read("$path.xz"))
+        } catch (e: FileNotFoundException) {
+            // Fall back to try reading the uncompressed file.
+            super.read(path)
```

Comment:
Note the this changes the exceptions thrown by this function, e.g. if read throws `FileNotFoundException` it now points to `$path` instead of to `$path.xz`.

Reply:
> Note that the exceptions thrown by this function change, e.g. if `read()` throws `FileNotFoundException` it now points to `$path` instead of to `$path.xz`.

Good point, done.

> Did you consider instead of just `reading` the fallback file, to instead fix-it up by writing it to a `.xz` and then `deleting` the uncompressed file?

Yes, but I decided against it, as calling `read` should never have the side-effect of *writing* something, IMO. Also, if the user would switch back to uncompressed storage (for whatever reason), the original uncompressed files would be gone (but keeping both uncompressed and compressed files does not make either).

Reply:
Thanks!